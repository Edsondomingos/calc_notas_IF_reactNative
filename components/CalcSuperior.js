import * as React from 'react';
import { useState } from 'react';
import { Text, View, StyleSheet, TextInput, Button, TouchableHighlight } from 'react-native';
import Styles from '../assets/styles'

let finall

export default function CalcSuperior(){
  const [nota1, setNota1] = useState('')
  const [nota2, setNota2] = useState('')
  const [final, setFinal] = useState()
  const [resultado, setResultado] = useState()
  
  const Calcular = () => {
    let n1 = nota1
    let n2 = nota2
    if(nota1.length == 0){
      setResultado('Digite pelo menos a primeira nota')
    }
    if(nota1.length != 0 && nota2.length == 0){
      finall = (((300 - (n1*2))/3))
      setNota2(finall.toFixed(2))
      setResultado(`
      Calculo da média \n(${nota1.toFixed(2)} + ${finall.toFixed(2)}) / 5 = 60`)
    } 
    if (nota1.length != 0 && nota2.length != 0){
      finall = ((n1 * 2) + (n2 * 3)) / 5
      if(finall > 20 && finall < 60){
        if(nota1 > nota2){
          finall = (((300 - (n1*2))/3))
          setFinal(finall.toFixed(2))
          setResultado(`
          Voce precisa tirar: \n ${finall.toFixed(2)} na prova final`)
        } else if(nota2 > nota1){
          finall = (((300 - (n2*3))/2))
          setFinal(finall)
          setResultado(`
          Voce precisa tirar: \n ${finall.toFixed(2)} para ser aprovado`)
        } else {
          setResultado(`Você precisa tirar:\n ${120-finall.toFixed(2)} na prova final`)
        }
      } else if(finall < 20){
        setResultado(`Reprovado por média: ${finall.toFixed(2)}`)
      } else if(finall >= 60){
      setResultado(`APROVADO \nSua média foi: ${finall.toFixed(2)}`)
      }
    }
    
  }

  function Limpar(){
    setNota1('')
    setNota2('')
    setFinal('')
    setResultado('')
  }


  return (
    <View style={Styles.container}>
        <Text style={Styles.titulo}>Calculadora de Notas cursos Superiores</Text>
        <View style={Styles.ContainerNotas}>
          <Text style={Styles.textoNotas}>N1</Text>
          <TextInput 
            style={Styles.input}
            placeholder='Digite a primeira nota'
            value={nota1}
            onChangeText={setNota1}
            testID='inputN1'
          />

          <Text style={Styles.textoNotas}>N2</Text>
          <TextInput 
             style={Styles.input}
             placeholder='Digite a segunda nota'
             value={nota2}
             onChangeText={setNota2}
             testID='inputN2'
          />

          <Text style={Styles.textoNotas}>FINAL</Text>
          <TextInput 
             style={Styles.input}
             value={final}
             onChangeText={setFinal}
             testID='inputFinal'
          />

          <View style={Styles.botoes}>
            <TouchableHighlight 
                style={Styles.btCalcular}
                onPress={() => Limpar()}
                testID='btnLimpar'
              >
                <Text style={Styles.textoBt}>Limpar</Text>
              </TouchableHighlight>

              <TouchableHighlight 
                style={Styles.btCalcular}
                onPress={() => Calcular()}
                testID='btnCalcular'
              >
                <Text style={Styles.textoBt}>Calcular</Text>
              </TouchableHighlight>

            
          </View>
          <Text style={Styles.resultado} testID='resultado'>{resultado}</Text>
        
        </View>
    </View>
  )
}