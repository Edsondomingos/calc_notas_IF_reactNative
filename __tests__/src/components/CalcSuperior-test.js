import renderer from 'react-test-renderer'
import { render, fireEvent } from "@testing-library/react-native";
import CalcSuperior from '../../../components/CalcSuperior';
jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

describe('Teste de snapshot', () => {
    it('Snapshot de tela CalcSuperior', () => {
        const tela = renderer.create(<CalcSuperior />).toJSON()
        expect(tela).toMatchSnapshot()
    })    
})

describe('Teste de componente tela Home', () => {
    it('BotÃ£o Superior', () => {
        const {getByText, getByTestId } = render(<CalcSuperior />)

        const titulo = getByText('Calculadora de Notas cursos Superiores')
        expect(titulo).toBeTruthy()

    })
})