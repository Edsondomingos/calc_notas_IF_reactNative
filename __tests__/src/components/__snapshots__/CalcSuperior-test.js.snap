// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot Snapshot de tela CalcSuperior 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Cursive",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginVertical": 20,
        "textAlign": "center",
      }
    }
  >
    Calculadora de Notas cursos Superiores
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      N1
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Digite a primeira nota"
      style={
        Object {
          "borderRadius": 7,
          "borderWidth": 1,
          "margin": "auto",
          "marginBottom": 20,
          "padding": 5,
          "width": 200,
        }
      }
      value=""
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      N2
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Digite a segunda nota"
      style={
        Object {
          "borderRadius": 7,
          "borderWidth": 1,
          "margin": "auto",
          "marginBottom": 20,
          "padding": 5,
          "width": 200,
        }
      }
      value=""
    />
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      FINAL
    </Text>
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderRadius": 7,
          "borderWidth": 1,
          "margin": "auto",
          "marginBottom": 20,
          "padding": 5,
          "width": 200,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#21AEEF",
            "borderRadius": 20,
            "height": 45,
            "marginHorizontal": 10,
            "marginVertical": 20,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Limpar
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#21AEEF",
            "borderRadius": 20,
            "height": 45,
            "marginHorizontal": 10,
            "marginVertical": 20,
            "width": 120,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 25,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Calcular
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#f00",
          "fontSize": 25,
          "fontWeight": "bold",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
